receivers/mqtt/config.go

```diff
@@ -16,14 +16,19 @@ const (
 )
 
 type Config struct {
-	BrokerURL          string `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
-	ClientID           string `json:"clientId,omitempty" yaml:"clientId,omitempty"`
-	Topic              string `json:"topic,omitempty" yaml:"topic,omitempty"`
-	Message            string `json:"message,omitempty" yaml:"message,omitempty"`
-	MessageFormat      string `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
-	Username           string `json:"username,omitempty" yaml:"username,omitempty"`
-	Password           string `json:"password,omitempty" yaml:"password,omitempty"`
-	InsecureSkipVerify bool   `json:"insecureSkipVerify,omitempty" yaml:"insecureSkipVerify,omitempty"`
+	BrokerURL            string                   `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
+	ClientID             string                   `json:"clientId,omitempty" yaml:"clientId,omitempty"`
+	Topic                string                   `json:"topic,omitempty" yaml:"topic,omitempty"`
+	Message              string                   `json:"message,omitempty" yaml:"message,omitempty"`
+	MessageFormat        string                   `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
+	Username             string                   `json:"username,omitempty" yaml:"username,omitempty"`
+	Password             string                   `json:"password,omitempty" yaml:"password,omitempty"`
+	QoS                  receivers.OptionalNumber `json:"qos,omitempty" yaml:"qos,omitempty"`
```

Comment:
Let's use int. IIRC OptionalNumber was introduced to support backward compatibility with integrations from legacy alerting.